### PR TITLE
Cache translation

### PR DESCRIPTION
**Pull request by c0b41**

### CODE REVIEW

File: repositories/translationRepository.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
 import LocaleString from "../entities/LocaleString";
 
+const locale_cache = new Map();
```
Comment (axross) [inline repositories/translationRepository.ts]:
```suggestion
const localeCache = new Map();
```

File: repositories/translationRepository.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+
 async function getTranslation(
   locale: LocaleString
 ): Promise<Record<string, string>> {
+    
+  if (typeof window !== "undefined" && locale_cache.has(locale)) {
+    return locale_cache.get(locale)
+  }
```
Comment (axross) [inline repositories/translationRepository.ts]: This block is supposed to be inside the next `if (typeof window !== "undefined") { ... }`.

```typescript
if (typeof window !== "undefined") {
    if (localeCache.has(locale)) {
        return localeCache.get(locale)
    }

    const response = await fetch(`/static/translation/${locale}.json`); 
    // ...
}
```

File: repositories/translationRepository.ts [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+    
+  if (typeof window !== "undefined" && locale_cache.has(locale)) {
+    return locale_cache.get(locale)
+  } 
+   
   if (typeof window !== "undefined") {
     const response = await fetch(`/static/translation/${locale}.json`);
     const json = await response.json();
-
+    locale_cache.set(locale, json)
```
Comment (axross) [inline repositories/translationRepository.ts]:
```suggestion
    localeCache.set(locale, json)
```

**Comment by axross** [comment]
@c0b41 Thank you! Is it ready?

**Comment by c0b41** [comment]
@axross yea, ready the go

**Comment by axross** [comment]
Thanks a lot!

**Pull request merged**